### ISSUE TITLE
AfterEach doesn't run if a test fails due to unhandled exception

### ISSUE THREAD

**Issue by danger-ranger** [paulcwarren/ginkgo4j#3]

**Comment by danger-ranger** [comment]
Judging by this line it appears that this is intentional:
https://github.com/paulcwarren/ginkgo4j/blob/2403e9c45dfc58de819db841dff3f86fae9a689e/src/test/java/impl/com/github/paulcwarren/ginkgo4j/runner/SpecRunnerThreadTests.java#L72

However it feels wrong. In my opinion the only AfterEach's that should be ignored are the ones at or below the context of the failing BeforeEach. 

Specifically, if I have an AfterEach at the top level of my test contexts and deep down in a Describe a BeforeEach fails I still expect the top level AfterEach to be run.

**Comment by danger-ranger** [comment]
After consulting the rspec docs I think they support our interpretation:
https://www.relishapp.com/rspec/rspec-core/v/2-2/docs/hooks/before-and-after-hooks#failure-in-before(:all)-block

specifically:

```
an error in before(:all)
  fails this example (FAILED - 1)
  fails this example, too (FAILED - 2)
  nested group
    fails this third example (FAILED - 3)
    fails this fourth example (FAILED - 4)
    yet another level deep
      fails this last example (FAILED - 5)
after all ran
```

**Comment by paulcwarren** [comment]
Hi Gavin,

I didn't religiously check my implementation against the RSpec spec as I was more concerned about getting something out there.  

For the scenario you highlight the RSpec behavior seems entirely sensible and I will look into this for you.  If you already have a fix then I would gladly accept a PR as well.  

Thanks for bringing this up.

**Comment by paulcwarren** [comment]
I just tested ginkgo as well and that performs as per the spce so this definitely needs addressing.  I'll put at the top of my backlog and look at it immediately.

**Comment by danger-ranger** [comment]
@dprotaso is on our team and he's familiar with the code base so we might get to it before you :)

**Comment by paulcwarren** [comment]
Fixed in 1.0.4
